extensions/resteasy-reactive/quarkus-rest-client-microprofile/runtime/src/main/java/io/quarkus/resteasy/reactive/client/microprofile/MicroProfileRestRequestClientFilter.java

```diff
@@ -0,0 +1,96 @@
+package io.quarkus.resteasy.reactive.client.microprofile;
+
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.annotation.Priority;
+import javax.enterprise.context.RequestScoped;
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientRequestFilter;
+import javax.ws.rs.core.MultivaluedHashMap;
+import javax.ws.rs.core.MultivaluedMap;
+
+import org.eclipse.microprofile.rest.client.ext.ClientHeadersFactory;
+import org.eclipse.microprofile.rest.client.ext.DefaultClientHeadersFactoryImpl;
+
+import io.quarkus.arc.Arc;
+import io.smallrye.common.constraint.NotNull;
+import io.smallrye.common.constraint.Nullable;
+
+@Priority(Integer.MIN_VALUE)
+public class MicroProfileRestRequestClientFilter implements ClientRequestFilter {
+    private static final MultivaluedMap<String, String> EMPTY_MAP = new MultivaluedHashMap<>();
+
+    @Nullable
```

Comment:
Do we really want these annotations around? I haven't seen there elsewhere in the codebase

Reply:
It's used from generated code only, so it's not really easy to check how it's used. And one of the params cannot be null while other can. I can change it to a comment if that would fit how it's currently done better.

Reply:
I don't really mind, I am just wondering as I haven't seen those SR annotations anywhere else